app/controllers/carts_controller.rb

```diff
@@ -27,8 +27,35 @@ def create
     redirect_to items_path
   end
 
+  def update
+    @item = Item.find(params[:item_id])
+    @item_id_str = @item.id.to_s
+    @new_quantity = params[:quantity].to_i || 0
+    @current_quantity = @cart.contents[@item_id_str].to_i
+    adjust_quantity
+    flash[:success] = "You now have #{pluralize(@new_quantity, "copy")} of #{@item.name} in your cart!"
+    redirect_to cart_path
+  end
+
+  def destroy
+    @cart.contents.clear
+    redirect_to cart_path
+  end
+
   private
 
+  def adjust_quantity
+    if @new_quantity > 0
```

Comment:
Couldn't determine a better scheme for handling the multitude of operations, so diverted logic to a helper method here to handle the conditionals for updating the cart. If anyone has a better Idea i'm open to it.

Reply:
I think it's fine, we went over this together